### PR TITLE
Dm 3256

### PR DESCRIPTION
**Pull request by JoshingYou1**
### JIRA issue link
https://agile6.atlassian.net/browse/DM-3256

## Description - what does this code do?
- Adds an alt text field for all image resource types for each practice resource (problem, solution, results, multimedia).
- Adds an alt text field for a practice's `main_display_image`.
- Changes label text for caption fields and add some text to image input labels.

### Prerequisite
In a terminal, run `rails db:migrate`.
### Main Display Image Alt Text
1. Log in as an admin and visit the introduction page of a practice of your choice.
2. Scroll down to the "Thumbnail" section and make sure the alt text field is now available.
3. Add some text to it and save your progress. Go to the show page of that practice and make sure the alt text for the image matches what you added to the field.
4. Remove the image, keep the alt text, and save your progress. Make sure the alt text field is cleared after saving if no image is present.
5. Now add only alt text with no image, save your progress, and make sure the alt text is not saved.
### Practice Resource Image Alt Text
1. Visit the overview page for the same practice.
2. Add multiple image resources for any resource area (problem, solution, results, multimedia) without adding alt text to them, and then save your progress.
3. Visit the practice show page and make sure each one has the "alt" attribute, but with no value.
4. Go back to the overview page and add alt text to any number of the uploaded images and save your progress.
5. Visit the practice show page and make sure the images you added alt text to in the editor now alt text that matches what you added in the editor.
### Styling
1. Make sure the "Caption" and alt text labels both match the designs in Figma (alt text fields were changed to textareas after the fact).
2. Make sure the main display image and image resource file inputs have labels that mention avoiding the use of images with text, just like in the Figma file.

## Link to mock-ups/mock ups (image file if you have it) (if applicable)
https://www.figma.com/file/ccwLbt94U9QGfnuFCwOXzY/Core-Designs---VA-DM?node-id=680%3A1021

## Acceptance criteria
- Add Alt Text field 
- Add Alt Text explainer text
- Add Caption explainer text
- Add “Please Avoid using images that are only text”
- Make tweaks in design
- Make sure “Caption” is used consistently 
- Alt Text is NOT a required field

## Definition of done
- [X] Unit tests written (if applicable)
- [X] e2e/accessibility tests written (if applicable)
- [X] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [X] A link has been provided to the originating JIRA issue
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by camillevilla — CHANGES_REQUESTED** [review]
Hey Josh! I think the technical implementation in the editor is working great. Thanks for the detailed tests too.

I have some reservations about adding default alt text to images where the content creator did not provide alt text. I threw in some images and GIFs to illustrate what a screenreader user would hear. In most cases, the default alt text would add noise and would be less descriptive than the already required image captions. I know we're in a tough spot with existing data and bringing all images on the site up to compliance. If leaving out alt text isn't an option, I'd advocate using the least verbose option possible. If there's context I'm missing from the JIRA story or earlier discussions, let me know! I'm happy to hop on a call.

**Review by bradjohnson92008 — APPROVED** [review]
Everything worked as intended for me.  Great work Josh.

File: app/views/practices/show/_main_display_image.html.erb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <% if @practice.main_display_image.present? %>
+  <% main_image_alt_text = @practice.main_display_image_alt_text %>
   <img
     class="dm-thumbnail-image display-block"
     src="<%= @practice.main_display_image_s3_presigned_url(:thumb) %>"
-    alt="<%= @practice.name %> main image"/>
+    alt="<%= main_image_alt_text.present? ? main_image_alt_text : "#{@practice.name} main image" %>"/>
```
[the remark this answers is not in the document]
Comment (JoshingYou1) [reply app/views/practices/show/_main_display_image.html.erb]: I went ahead and made it blank unless the user adds some! 😄

File: app/helpers/practice_resources_helper.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module PracticeResourcesHelper
+  def resource_image_default_alt_text(resource_record)
+    resource_class_name = resource_record.class.to_s.split(/(?=[A-Z])/).second
+    revised_resource_class_name = resource_class_name === 'Multimedium' ? 'Multimedia' : resource_class_name
+    "#{revised_resource_class_name} image #{resource_record.position}"
+  end
+end
```
[the remark this answers is not in the document]
Comment (JoshingYou1) [reply app/helpers/practice_resources_helper.rb]: I removed all of the default alt text 😃

**Review by camillevilla — APPROVED** [review]

**Pull request merged**